Various things involving the Captain's Laser and display cases.

Rukofamicom opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR largely reverts #1213 because it is both a sloppy way to do what that PR did and also to resolve an issue where players don't understand they are not actually intended to just take the laser out of the case at the start of the round. The gun is intended as a trophy/display case item, not to be at the captain's side at all times. **The case is now locked to the captain until the station's alert level elevates to at least Blue**

Because it's being added to the captain's case anyway, I went ahead and made the functionality to create security-level dependent locks a part of the base display case object in code. Future display case implementations can make use of this, and there's even a possibility of introducing fireaxe-type wall cases to open for everyone during highly-elevated alert levels. 

In order to prevent a strange situation where a case would be locked in an open state, display cases no longer require ID access to close them - only to open. 

Due to the fact this supposedly 'antique' laser weapon is also one of the most powerful on the station, it has been somewhat nerfed. It is still a very powerful gun, but it is no longer a direct upgrade over the lethal-capable guns in the armory. 

This PR also fixes a visual bug that has persisted for a very long time: The captain's gun almost never displays its empty state even when it is currently out of charge and cannot fire another shot.

![CapGunEmpty](https://user-images.githubusercontent.com/9547572/165268964-71b7b5c1-c120-4a54-b31c-feeacc437343.gif) 

At the request of @Archanial, this PR now also includes a minor refactor to defining how much charge non-standard guns have. It is now defined as a single variable when initializing the gun instead of hacky manipulation of the power cell in each individual guns' code. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Players should not have unintuitively easy access to an item they are not only not intended to take, and even punished by admins for taking. The case will unlock for the captain after the alert level has been elevated, and if for some reason that isn't possible to do, he can always smash into the case in an emergency just like anyone else. 

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

Tested extensively. See past edits on PR because this section has been changed so many times I can barely keep track of it at this point lmao. 

</details>

## Changelog
:cl:
fix: Fixed a bug with the captain's laser which prevented it from appropriately displaying when the gun had no charge
balance: The captain's "antique" laser has been reduced to a 10 shot capacity, so that it is more representative of being an antique. 
tweak: The captain's laser is now appropriately locked behind its display case again to make it more intuitive that players are not intended to take the gun from the start of the round. The captain will be able to open this case during elevated alert levels.
tweak: All display cases may now be closed even if you do not have access to the case.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rukofamicom committed:
Cap gun bugfix + PR 1213 reverted

ivanmixo commented:
Never heard anyone ever have a problem with captain carrying the antique on him.

Archanial commented on _maps/map_files/PubbyStation/PubbyStation.dmm:
```diff
@@ -67,11 +67,7 @@
-	req_access_txt = "20";
-	req_one_access_txt = "0"
-	},
+/obj/structure/displaycase/captain
```
Missing ','

Rukofamicom replied:
Ah, dammit. Thank you.

Archanial commented on code/modules/projectiles/guns/energy/laser.dm:
```diff
@@ -30,14 +30,17 @@
 
 /obj/item/gun/energy/laser/captain
-	name = "antique laser gun"
+	name = "commemorative laser gun"
```
I personally think the name should remain unchanged

Rukofamicom replied:
An antique gun would be old, outdated and completely inferior to modern weapons. The captain's gun is not an antique at all unless Nanotrasen has failed to produce new technology for a very long time - it outshines everything in the armory at the start of each shift.

PowerfulBacon replied:
If its changing the traitor obj needs to be done too

Rukofamicom replied:
Ah whoops.

Archanial commented on code/modules/projectiles/guns/energy/laser.dm:
```diff
@@ -30,14 +30,17 @@
+	desc = "This is a commemorative laser gun issued by Nanotrasen as a reward for exemplary service. It is decorated with leather and chrome. On the side is an image of Space Station 13 with the words NTSSGolden engraved. All craftsmanship is of the highest quality." // Never forget Golden, even when changing this description. 
 	force = 10
 	ammo_x_offset = 3
+	cell_type = /obj/item/stock_parts/cell{charge = 700; maxcharge = 700} //This retain's the guns original capacity of 14 shots. Also just like before it will actually feel like 15 because at least one shot will recharge when discharging a full magazine
```
Make subtype of cell and instance object out of that instead of this.

On the other hand I had no idea dm supports that lol

Rukofamicom replied:
> On the other hand I had no idea dm supports that lol

This is how every energy gun with non-standard charge values does it. Take look at the code for the HoS laser, or the mini-energy gun. Doing something different here would make it different from every other similar instance in the code and be inconsistent.

Rukofamicom replied:
I have completely refactored how cell charge is defined in the latest commit

Archanial commented:
I see no problem with caps gun being locked again, although we should consider possibility of it unlocking during red or something.

Also, personally, I'd make the gun weaker. The idea of low ammo from previous pr was interesting.

Rukofamicom committed:
grumble syntax grumble comma

Rukofamicom commented:
> Also, personally, I'd make the gun weaker. The idea of low ammo from previous pr was interesting.

While this is the path I'd like to take since it works better on all ends in my opinon, Bacon very strongly disagrees. I'm not a big fan of the captain having a case he personally owns but can't unlock, and I'm not sure I'll be able to figure out how to make it automatically unlock at elevated alert (which is also something they recommended, and I mentioned I'd try to figure out at some point).

Rukofamicom committed:
update references to the antique gun

Rukofamicom committed:
refactors defining unique cell charge

Rukofamicom committed:
Fixes attachment overlays on guns

Sapfearon commented:
eh, i still think that captain unlocking case is good. Maybe allow to unlock it on red alert?

Coding-Grape commented:
>I'm not sure I'll be able to figure out how to make it automatically unlock at elevated alert (which is also something they recommended, and I mentioned I'd try to figure out at some point).

Couldn't you use parts of whatever handles the secure safe opening? Some of them only unlock off of green alert, not sure how well thst would mesh though 

Edit: Actually isn't there a global alert level variable that you can check to see if it's on red alert when they try to unlock the case? Might need to edit the case code a bit to add an alert level dependant locking system but it should be doable

Rukofamicom commented:
Definitely gives me a lead to look into for it, which usually ends up with results.

Rukofamicom committed:
Implements cases changing access levels at blue alert

Rukofamicom committed:
Fixes the display case in the lavaland snow biome

Captain being able to arbitrarily access this case would be a very strange occurrence.

Rukofamicom commented:
@PowerfulBacon I got the case to unlock appropriately at elevated security levels.

PowerfulBacon commented on code/game/objects/structures/displaycase.dm:
```diff
@@ -115,6 +117,8 @@
 
 /obj/structure/displaycase/attackby(obj/item/W, mob/user, params)
+	if(security_level_locked && GLOB.security_level != SEC_LEVEL_GREEN)
+		req_access = list(security_level_access)
```
Needs to be reset when alert is lowered again.

PowerfulBacon replied:
Also is there a signal you could register for alert level change, to update it on change?

Rukofamicom replied:
There is no signal I could find, no. 

Updated the case to reset when alert level is lowered. 
Also updated all cases to no longer require ID access to close, to prevent some unusual situations where cases can be permanently locked open, especially if someone decides to implement additional display cases with variable security locks at some point.

Rukofamicom committed:
Merge gunlights into this PR

Gunlights

Rukofamicom committed:
You can now close displays without access

Archanial commented on code/game/objects/structures/displaycase.dm:
```diff
@@ -14,6 +14,9 @@
 	var/openable = TRUE
+	var/security_level_locked = FALSE //For locking the case to different access levels based on whether the station is at an elevated security level
+	var/security_level_access = ACCESS_SECURITY	//Access level required after security level elevates
+	var/green_level_access	//for storing the original access requirements
```
Those 3 could just be one var

Rukofamicom replied:
Done and also simplified my overly complicated If statements.

Archanial commented on code/modules/projectiles/guns/energy.dm:
```diff
@@ -6,6 +6,7 @@
 
 	var/obj/item/stock_parts/cell/cell //What type of power cell this uses
 	var/cell_type = /obj/item/stock_parts/cell
+	var/gun_charge = 1000 // how much charge the cell will have, this is equal to the default power cell defined above
```
Put the comment above var with 3 `/`

Archanial commented on code/modules/projectiles/guns/energy/laser.dm:
```diff
@@ -30,14 +30,17 @@
+	gun_charge = 700 //This retain's the guns original capacity of 14 shots. Also just like before it will actually feel like 15 because at least one shot will recharge when discharging a full magazine
 	selfcharge = 1
+	charge_delay = 10 // Default is 8, setting this to 10 instead slightly slows the rate of recharge ticks, but the gun now recharges two shots per tick for an overall slight increase in regeneration.
+	ammo_type = list(/obj/item/ammo_casing/energy/lasergun/captain)
```
comments

Rukofamicom committed:
Merge branch 'BeeStation:master' into Case-closed

Rukofamicom committed:
Greatly cleans up the code

Rukofamicom committed:
One more comment updated

Rukofamicom committed:
Cap's gun is an antique again

Rukofamicom commented:
By request and with agreement from Bacon I have changed the captain's gun back into an antique, but also slightly weakened it compared to its original state. It is still a very strong gun, but no longer a direct upgrade over every lethal-capable gun in the armory. 

I believe the PR description is accurate now

Rukofamicom committed:
variables explained

Rukofamicom committed:
vend-a-tray stuff

Rukofamicom committed:
Merge branch 'master' into Case-closed

Rukofamicom committed:
Update displaycase.dm

Daridine commented:
> Never heard anyone ever have a problem with captain carrying the antique on him.

I think I've handled this at least once every other time I hop on to admin.

PestoVerde322 reviewed (changes requested):
![wrongcase](https://user-images.githubusercontent.com/75247747/166308869-a30a944d-0152-4b96-974d-97c730a5b40f.png)
Apparently, lavaland_surface_biodome_winter has a non-existant case, quite sure you planned to change it to /obj/structure/displaycase/captain or something else.

EDIT:  Apparently i forgot that i was working on my own branch, this means that it is good to go!

Archanial reviewed (changes requested):
Address review from @PestoVerde322

Rukofamicom commented:
There is nothing to address, his post says so at the bottom.  I talked to him on discord about it already. 

> EDIT: Apparently i forgot that i was working on my own branch, this means that it is good to go!

PestoVerde322 commented:
yeah i kinda slipped and forgot i was working on my own branch instead of Ruko's, I can confirm once again it should be all good to go!
